Sources/WebSocketKit/WebSocket.swift

```diff
@@ -88,12 +94,65 @@ public final class WebSocket {
         let string = String(text)
         var buffer = channel.allocator.buffer(capacity: text.count)
         buffer.writeString(string)
-        self.send(raw: buffer.readableBytesView, opcode: .text, fin: true, promise: promise)
 
+        self.send(buffer: buffer, opcode: .text, promise: promise)
     }
 
     public func send(_ binary: [UInt8], promise: EventLoopPromise<Void>? = nil) {
-        self.send(raw: binary, opcode: .binary, fin: true, promise: promise)
+        var buffer = channel.allocator.buffer(capacity: binary.count)
+        buffer.writeBytes(binary)
+        self.send(buffer: buffer, opcode: .binary, promise: promise)
+    }
+    
+    public func send(
+        buffer: NIO.ByteBuffer,
+        opcode: WebSocketOpcode,
+        promise: EventLoopPromise<Void>? = nil
+    ) {
+        guard buffer.readableBytes > outboundMaxFrameSize.value else {
+            let frame = WebSocketFrame(
+                fin: true,
+                opcode: opcode,
+                maskKey: self.makeMaskKey(),
+                data: buffer
+            )
+            self.channel.writeAndFlush(frame, promise: promise)
+            return
+        }
+        
+        var buffer = buffer
+        
+        var framesToSend: [WebSocketFrame] = []
+        
+        while let frameBuffer = buffer.readSlice(length: outboundMaxFrameSize.value) {
+            let frame = WebSocketFrame(
+                fin: buffer.readableBytes == 0,
+                opcode: opcode,
+                maskKey: self.makeMaskKey(),
+                data: frameBuffer
+            )
+            framesToSend.append(frame)
+        }
+        
+        if buffer.readableBytes > 0 {
+            let frame = WebSocketFrame(
+                fin: true,
+                opcode: opcode,
+                maskKey: self.makeMaskKey(),
+                data: buffer
+            )
+            framesToSend.append(frame)
+        }
+        
+        let startingOut: EventLoopFuture<Void> = self.channel.eventLoop.makeSucceededFuture(Void())
+        
+        let future: EventLoopFuture<Void> = framesToSend.reduce(startingOut) { future, frame in
+            return future.flatMap { _ in
+                self.channel.writeAndFlush(frame)
+            }
+        }
```

Comment:
im not 100% sure what is the best way to do this sending of multiple frames but i figured the order is important so sending the next frame only after the last frame was sent is best?